src/common/src/main/scala/osmesa/common/raster/package.scala

```diff
@@ -0,0 +1,129 @@
+package osmesa.common
+import geotrellis.macros.IntTileVisitor
+import geotrellis.raster.{
+  ArrayTile,
+  CellType,
+  IntCellType,
+  IntCells,
+  IntConstantNoDataCellType,
+  IntUserDefinedNoDataCellType,
+  MutableArrayTile,
+  NoDataHandling,
+  Tile
+}
+
+import scala.collection.mutable
+
+package object raster {
+  class SparseIntTile(val cols: Int,
+                      val rows: Int,
+                      val values: Map[Long, Int],
+                      val cellType: IntCells with NoDataHandling)
+      extends ArrayTile {
+    private val noDataValue = cellType match {
+      case IntConstantNoDataCellType        => Int.MinValue
+      case IntUserDefinedNoDataCellType(nd) => nd
+      case IntCellType                      => 0
+    }
+
+    def interpretAs(newCellType: CellType): Tile = {
+      newCellType match {
+        case dt: IntCells with NoDataHandling =>
+          SparseIntTile(cols, rows, values, dt)
+        case _ =>
+          withNoData(None).convert(newCellType)
+      }
+    }
+
+    def withNoData(noDataValue: Option[Double]): Tile =
+      SparseIntTile(cols, rows, values, cellType.withNoData(noDataValue))
+
+    override def applyDouble(i: Int): Double = apply(i).toDouble
+
+    override def apply(i: Int): Int = values.getOrElse(i, 0)
+
+    override def copy: ArrayTile = SparseIntTile(cols, rows, Map(values.toSeq: _*), cellType)
+
+    override def toBytes(): Array[Byte] = ???
```

Comment:
Unimplemented because it doesn't make sense (and isn't used by code that uses `SparseIntTile`).